crates/rpc/rpc/src/eth/api/transactions.rs

```diff
@@ -1298,14 +1298,17 @@ mod tests {
     use reth_node_ethereum::EthEvmConfig;
     use reth_primitives::{constants::ETHEREUM_BLOCK_GAS_LIMIT, hex_literal::hex, Bytes};
     use reth_provider::test_utils::NoopProvider;
-    use reth_transaction_pool::{test_utils::testing_pool, TransactionPool};
+    use reth_transaction_pool::{
+        test_utils::{TestPool, TestPoolWrapper},
+        TransactionPool,
+    };
 
     #[tokio::test]
     async fn send_raw_transaction() {
         let noop_provider = NoopProvider::default();
         let noop_network_provider = NoopNetwork::default();
 
-        let pool = testing_pool();
+        let pool: TestPool = TestPoolWrapper::default().into();
```

Comment:
here we don't gain anything, so I'd prefer keeping the function where possible

Reply:
Yes, I agree, it was to prevent a function from wandering outside the structure, but you're right, it's better that way. I modified all the places where we had to enforce the type, tell me if that suits you?